### PR TITLE
feat: [PLG-2860]: canny autthenitication using canny sdk

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 1]
```diff
@@ -28,9 +28,8 @@ export const timestamp = moment.now()
 export const HARNESS_SUPPORT_LINK =
   '/sso.html?action=login&brand_id=114095000394&locale_id=1&return_to=https%3A%2F%2Fsupport.harness.io%2Fhc%2Fen-us%2Frequests&src=zendesk&timestamp=' +
   timestamp
-export const CANNY_SUPPORT_LINK =
-  '/gateway/sso.html?src=canny&action=login&companyID=6450229969594e0bc28b76ef&redirect=https%3A%2F%2Fharness-io.canny.io%2F'
-
+export const CANNY_SUPPORT_LINK = 'https://ideas.harness.io/'
+const appId = '6450229969594e0bc28b76ef'
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:32]: this should come from the env config, to allow overriding and avoid hardcoding

File: src/index.html [hunk 2]
```diff
@@ -249,6 +250,37 @@
...
+    <script>
+      if (window.CANNY_ENABLE) {
+        !(function (w, d, i, s) {
+          function l() {
+            if (!d.getElementById(i)) {
+              var f = d.getElementsByTagName(s)[0],
+                e = d.createElement(s)
+              ;(e.type = 'text/javascript'),
+                (e.async = !0),
+                (e.src = 'https://canny.io/sdk.js'),
```
Comment (abhinavrastogi-harness) [inline src/index.html:263]: this origin needs to be whitelisted in the CSP policy
https://github.com/harness/harness-core-ui/blob/develop/docker/entrypoint.sh#L57

File: src/index.html [hunk 3]
```diff
@@ -176,6 +176,7 @@
       window.HARNESS_ENABLE_APPDY_EUM = 'HARNESS_ENABLE_APPDY_EUM_PLACEHOLDER' === 'true'
       window.HARNESS_ENABLE_CDN = 'HARNESS_ENABLE_CDN_PLACEHOLDER' === 'true'
       window.HARNESS_ENABLE_SABER = 'HARNESS_ENABLE_SABER_PLACEHOLDER' === 'true'
+      window.CANNY_ENABLE = 'CANNY_ENABLE_PLACEHOLDER' === 'true'
```
Comment (abhinavrastogi-harness) [inline src/index.html:179]: can we follow the same pattern as above third-party integrations?

`window.HARNESS_ENABLE_CANNY`

File: docker/entrypoint.sh [hunk 4]
```diff
@@ -16,6 +16,7 @@ sed -i "s|<\!-- IDPUrl -->|<script>window.IDPUrl = '$IDP_URL'</script>|" index.h
 sed -i "s|HARNESS_ENABLE_NG_AUTH_UI_PLACEHOLDER|$HARNESS_ENABLE_NG_AUTH_UI_PLACEHOLDER|" index.html
 sed -i "s|HARNESS_BROWSER_ROUTER_ENABLED|$HARNESS_BROWSER_ROUTER_ENABLED|" index.html
 sed -i "s|HARNESS_NO_AUTH_HEADER|$HARNESS_NO_AUTH_HEADER|" index.html
+sed -i "s|CANNY_ENABLE|$CANNY_ENABLE|" index.html
```
Comment (abhinavrastogi-harness) [inline docker/entrypoint.sh:19]: you are using the variable `CANNY_ENABLE_PLACEHOLDER` in HTML, but a defining a different variable here. it'll always be undefined.

Also, can we use the same pattern as other third-party integrations?

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 5]
```diff
@@ -40,9 +39,30 @@ export const openZendeskSupport = (e: React.MouseEvent<Element, MouseEvent>): vo
...
 
-export const cannySupportShareYourIdeas = (e: React.MouseEvent<Element, MouseEvent>): void => {
+const initCanny = (currentUserInfo: UserInfo) => {
+  Canny('identify', {
+    appID: appId,
+    user: {
+      companies: [
+        {
+          id: currentUserInfo.accounts?.[0].companyName,
+          name: currentUserInfo.accounts?.[0].companyName
```
Comment (kapil-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:49]: why to pick first account from a list of account the user is part of? rather we should pick the account in which the user is logged into.

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 6]
```diff
@@ -40,9 +39,30 @@ export const openZendeskSupport = (e: React.MouseEvent<Element, MouseEvent>): vo
...
 }
 
-export const cannySupportShareYourIdeas = (e: React.MouseEvent<Element, MouseEvent>): void => {
+const initCanny = (currentUserInfo: UserInfo) => {
+  Canny('identify', {
+    appID: appId,
+    user: {
+      companies: [
+        {
+          id: currentUserInfo.accounts?.[0].companyName,
```
Comment (kapil-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:48]: companyId should be accountId, reference slack thread: https://harness.slack.com/archives/C05CKH0G10W/p1696549500494369

**Review by kapil-harness — CHANGES_REQUESTED** [review]

**Comment by dhruvUpadhyay07** [comment]
retrigger cypress-pipeline

File: docker/entrypoint.sh [hunk 7]
```diff
@@ -37,6 +38,8 @@ sed -i "s|<\!-- newNavContentfulEnvironment -->|<script>window.newNavContentfulE
 sed -i "s|<\!-- harnessNameSpacePlaceHolder -->|<script>window.harnessNameSpace = '$HARNESS_NAME_SPACE'</script>|" index.html
 sed -i "s|<\!-- harnessClusterURLPlaceHolder -->|<script>window.harnessClusterURL = '$HARNESS_CLUSTER_URL'</script>|" index.html
 sed -i "s|<\!-- stripeApiKey -->|<script>window.stripeApiKey = '$STRIPE_API_KEY'</script>|" index.html
+sed -i "s|<\!-- harnessAppId -->|<script>window.harnessAppId = '$HARNESS_APP_ID'</script>|" index.html
```
Comment (abhinavrastogi-harness) [inline docker/entrypoint.sh:41]: we should use the thirdparty name here.. call it `cannyAppId`

File: docker/entrypoint.sh [hunk 8]
```diff
@@ -54,7 +57,7 @@ fi
...
+  sed -i "s|#cspHeadersPlaceholder|add_header Content-Security-Policy-Report-Only \"script-src 'self' 'unsafe-inline' https://canny.io/sdk.js https://*.harness.io https://widget.saberfeedback.com https://cdn.segment.com https://js.refiner.io https://widget.intercom.io https://js.intercomcdn.com https://d2wy8f7a9ursnm.cloudfront.net https://cdn.appdynamics.com; style-src 'self' 'unsafe-inline' https://*.harness.io https://fonts.googleapis.com https://widget.saberfeedback.com https://js.refiner.io; img-src 'self' data: blob: https://*.harness.io https://widget.saberfeedback.com; font-src 'self' https://fonts.gstatic.com;\";|" $NGINX_CONFIG_FILE
```
Comment (abhinavrastogi-harness) [inline docker/entrypoint.sh:60]: CSP should have only the origins we want to connect to.
adding a filepath like `/sdk.js` will not work.

File: src/modules/10-common/components/ResourceCenter/ResourceCenter.tsx [hunk 9]
```diff
@@ -56,6 +56,9 @@ export const ResourceCenter: React.FC<ResourceCenterProps> = ({ link, isOpen: di
   const [buttonDisabled, setButtonDisabled] = useState(false)
   const [show, setShow] = useState<boolean>(Boolean(display))
   const { showModal } = useReleaseNotesModal()
+  const { pathname } = useLocation()
+  const { accounts } = currentUserInfo
+  const account = accounts?.find(({ uuid }) => pathname.includes(uuid as string)) || {}
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/ResourceCenter.tsx:61]: we already have accountInfo in AppStoreContext. We don't need to iterate over this list to extract it.

File: src/modules/10-common/components/ResourceCenter/utils.ts [hunk 10]
```diff
@@ -6,7 +6,7 @@
  */
 
 import moment from 'moment'
-import type { UserInfo } from 'services/cd-ng'
+import type { GatewayAccountRequestDTO, UserInfo } from 'services/cd-ng'
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/ResourceCenter/utils.ts:9]: This is not the correct DTO to use (1. it's gateway account, 2. it's a request DTO).

Use `AccountDTO`, which is the same as used in AppStoreContext.

File: docker/entrypoint.sh [hunk 11]
```diff
@@ -37,6 +38,8 @@ sed -i "s|<\!-- newNavContentfulEnvironment -->|<script>window.newNavContentfulE
 sed -i "s|<\!-- harnessNameSpacePlaceHolder -->|<script>window.harnessNameSpace = '$HARNESS_NAME_SPACE'</script>|" index.html
 sed -i "s|<\!-- harnessClusterURLPlaceHolder -->|<script>window.harnessClusterURL = '$HARNESS_CLUSTER_URL'</script>|" index.html
 sed -i "s|<\!-- stripeApiKey -->|<script>window.stripeApiKey = '$STRIPE_API_KEY'</script>|" index.html
+sed -i "s|<\!-- cannyAppId -->|<script>window.cannyAppId = '$HARNESS_APP_ID'</script>|" index.html
```
Comment (abhinavrastogi-harness) [inline docker/entrypoint.sh:41]: the RHS is still called `HARNESS_APP_ID`.. this should also be `CANNY_APP_ID`.

**Review by devinder-harness — APPROVED** [review]

**Pull request merged**